integration/singularity/store.go

```diff
@@ -431,3 +436,114 @@ func (s *SingularityStore) Describe(ctx context.Context, id blob.ID) (*blob.Desc
 	}
 	return descriptor, nil
 }
+
+func (s *SingularityStore) runCleanupWorker(ctx context.Context) {
+	// Run immediately once before starting ticker
+	go func() {
```

Comment:
To avoid false positive "cleanup already running" errors add wait group to assure that ticker triggered cycles never start until the first cycle finishes?

Reply:
i considered a mutex. the reason i did it this way is because if the cleanup worker really is running too slowly to keep up, it doesn't really make sense to have multiple goroutines up blocked on a mutex waiting for their turn to do a cleanup. 

i don't consider this situation to happen much, if ever, because it should never take so long, but it would log this error to the user if it happens as it's probably indicating something abnormal.

if it's really an issue, i think a better solution would be to add an additional atomic "retryCleanupImmediately" which can get set to true if a cleanup fails due to one already running, and it would trigger a new cleanup job immediately after the previous one finishes, and guarantees the maximum possible rate without possibly building up lots of blocked goroutines?